src/components/modals/FormStepSaveModal.js

```diff
@@ -50,11 +63,52 @@ const FormStepSaveModal = ({
   onSessionDestroyed,
   suspendFormUrl,
   suspendFormUrlLifetime,
+  ...props
 }) => {
   const intl = useIntl();
 
   const [{errorMessage, isSaving}, dispatch] = useImmerReducer(reducer, initialState);
 
+  const errorMap = (issue, ctx) => {
+    switch (issue.code) {
+      case z.ZodIssueCode.invalid_type: {
+        if (issue.received === z.ZodParsedType.undefined) {
+          const fieldName = issue.path.join('.');
+          const fieldLabel = intl.formatMessage(FIELD_LABELS[fieldName]);
+          const message = intl.formatMessage(
+            {
+              description: 'Required field error message',
+              defaultMessage: '{field} is required.',
+            },
+            {
+              field: fieldLabel,
+            }
+          );
+          return {message};
+        }
+        break;
+      }
+      case z.ZodIssueCode.invalid_string: {
+        const fieldName = issue.path.join('.');
+        const fieldLabel = intl.formatMessage(FIELD_LABELS[fieldName]);
+        const message = intl.formatMessage(
+          {
+            description: 'Invalid field value message',
+            defaultMessage: 'Invalid {field}.',
+          },
+          {
+            field: fieldLabel,
+          }
+        );
```

Comment:
I think you can specifically find problems with email fields here to give a better error message. See https://github.com/open-formulieren/formio-builder/blob/main/src/registry/email/edit-validation.ts#L20 for example